extension/intents/nicknames/nicknames.js

```diff
@@ -99,42 +99,18 @@ intentRunner.registerIntent({
 intentRunner.registerIntent({
   name: "nicknames.namePage",
   async run(context) {
-    const name = context.slots.name.toLowerCase();
+    const name = context.slots.name;
     const activeTab = await browserUtil.activeTab();
     const metadata = await pageMetadata.getMetadata(activeTab.id);
-    const result = await browser.storage.sync.get("pageNames");
-    const pageNames = result.pageNames || {};
-    pageNames[name] = metadata.url;
-    log.info("Added page name", name);
-    await browser.storage.sync.set({ pageNames });
+    intentRunner.registerPageName(name, metadata);
   },
 });
 
 intentRunner.registerIntent({
   name: "nicknames.removePageName",
   async run(context) {
-    const result = await browser.storage.sync.get("pageNames");
-    const pageNames = result.pageNames || {};
-    const getName = async () => {
-      const activeTab = await browserUtil.activeTab();
-      const metadata = await pageMetadata.getMetadata(activeTab.id);
-      return Object.keys(pageNames).find(
-        key => pageNames[key] === metadata.url
-      );
-    };
-    const name = context.slots.name || (await getName());
-    if (!name) {
-      const exc = new Error("No page name to remove");
-      exc.displayMessage = `This page does not have a name`;
-      throw exc;
-    }
-    if (!pageNames[name]) {
-      const exc = new Error("No page name to remove");
-      exc.displayMessage = `No page name "${name}" found`;
-      throw exc;
-    }
-    delete pageNames[name];
-    log.info("Removed page name", name);
-    await browser.storage.sync.set({ pageNames });
+    const name = context.slots.name;
+    await intentRunner.getRegisteredPageName(name);
+    intentRunner.registerPageName(name, { url: null });
```

Comment:
I guess this works, but it doesn't make it clear what's happening. A null URL doesn't really imply the page will be unregistered.

I guess it would be best to just create another function `intentRunner.unregisterPageName(name)`